core/core-funcs.el

```diff
@@ -49,7 +49,7 @@ FILE-TO-LOAD is an explicit file to load after the installation."
 
 (defun spacemacs//get-package-directory (pkg)
   "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (concat user-emacs-directory "elpa/")))
+  (let ((elpa-dir package-user-dir))
```

Comment:
The fact that you are potentially removing trailing slash has no impact in this case, because elpa-dir is used only in contexts where this does not matter.